L1Trigger/L1TTrackMatch/plugins/L1TkElectronTrackProducer.cc

```diff
@@ -0,0 +1,340 @@
+          iConfig.getParameter<edm::InputTag>("L1TrackInputTag"))) {
+  label = iConfig.getParameter<std::string>("label");  // label of the collection produced
+                                                       // e.g. EG or IsoEG if all objects are kept
+                                                       // EGIsoTrk or IsoEGIsoTrk if only the EG or IsoEG
+                                                       // objects that pass a cut RelIso < IsoCut are written
+                                                       // in the new collection.
+
+  ETmin = (float)iConfig.getParameter<double>("ETmin");
+
+  // parameters for the calculation of the isolation :
+  PTMINTRA = (float)iConfig.getParameter<double>("PTMINTRA");
+  DRmin = (float)iConfig.getParameter<double>("DRmin");
+  DRmax = (float)iConfig.getParameter<double>("DRmax");
+  DeltaZ = (float)iConfig.getParameter<double>("DeltaZ");
+  maxChi2IsoTracks = iConfig.getParameter<double>("maxChi2IsoTracks");
+  minNStubsIsoTracks = iConfig.getParameter<int>("minNStubsIsoTracks");
+  // cut applied on the isolation (if this number is <= 0, no cut is applied)
+  IsoCut = (float)iConfig.getParameter<double>("IsoCut");
+  RelativeIsolation = iConfig.getParameter<bool>("RelativeIsolation");
+
+  // parameters to select tracks to match with L1EG
+  trkQualityChi2 = (float)iConfig.getParameter<double>("TrackChi2");
+  trkQualityPtMin = (float)iConfig.getParameter<double>("TrackMinPt");
+  useTwoStubsPT = iConfig.getParameter<bool>("useTwoStubsPT");
+  dPhiCutoff = iConfig.getParameter<std::vector<double> >("TrackEGammaDeltaPhi");
+  dRCutoff = iConfig.getParameter<std::vector<double> >("TrackEGammaDeltaR");
+  dEtaCutoff = iConfig.getParameter<std::vector<double> >("TrackEGammaDeltaEta");
+  matchType = iConfig.getParameter<std::string>("TrackEGammaMatchType");
+
+  produces<TkElectronCollection>(label);
+}
+
+L1TkElectronTrackProducer::~L1TkElectronTrackProducer() {}
+
+// ------------ method called to produce the data  ------------
+void L1TkElectronTrackProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  std::unique_ptr<TkElectronCollection> result(new TkElectronCollection);
+
+  // geometry needed to call pTFrom2Stubs
+  edm::ESHandle<TrackerGeometry> geomHandle;
+  iSetup.get<TrackerDigiGeometryRecord>().get("idealForDigi", geomHandle);
+  const TrackerGeometry* tGeom = geomHandle.product();
+
+  // the L1EGamma objects
+  edm::Handle<EGammaBxCollection> eGammaHandle;
+  iEvent.getByToken(egToken, eGammaHandle);
+  EGammaBxCollection eGammaCollection = (*eGammaHandle.product());
+  EGammaBxCollection::const_iterator egIter;
+
+  // the L1Tracks
+  edm::Handle<L1TTTrackCollectionType> L1TTTrackHandle;
+  iEvent.getByToken(trackToken, L1TTTrackHandle);
+  L1TTTrackCollectionType::const_iterator trackIter;
+
+  if (!eGammaHandle.isValid()) {
+    edm::LogError("L1TkElectronTrackProducer") << "\nWarning: L1EmCollection not found in the event. Exit" << std::endl;
```

Comment:
should code execution stop in this case? if so, throw an exception

Reply:
yes. done.